core/src/apps/webauthn/confirm.py

```diff
@@ -22,14 +22,21 @@ def load_icon(self, rp_id_hash: bytes) -> None:
         from trezor import res
         from apps.webauthn.knownapps import knownapps
 
-        try:
-            namepart = knownapps[rp_id_hash]["label"].lower().replace(" ", "_")
-            icon = res.load("apps/webauthn/res/icon_%s.toif" % namepart)
-        except Exception as e:
-            icon = res.load("apps/webauthn/res/icon_webauthn.toif")
-            if __debug__:
-                log.exception(__name__, e)
-        self.app_icon = icon
+        app_name = knownapps.get(rp_id_hash, {}).get(
+            "label", None
+        )  # type: Optional[str]
+        if app_name is not None:
+            resource = "apps/webauthn/res/icon_%s.toif" % app_name.lower().replace(
+                " ", "_"
+            )
+            try:
+                self.app_icon = res.load(resource)
+            except Exception:
```

Comment:
Can we be here more specific about what Exception are we catching here?

Reply:
I'm working on this code now.
Should we treat `res.load` as if we expect it to succeed? Maybe it's a better option to add the resource path to `knownapps` directly so that the build process can tell which icons are present and which aren't?

Reply:
Looking at the `res` module, it looks like the only possibility would be `KeyError`.

Reply:
I'll change `except Exception:` to `except KeyError:` directly in master.

Reply:
Done https://github.com/trezor/trezor-firmware/commit/7c39e2f14239e1d51849c44a79d2471c1e62543b.